src/LanguageServer/Protocol/ExternalAccess/Razor/GetSymbolicInfoHandler.cs

```diff
@@ -0,0 +1,312 @@
+                                            .Any(variable => identifiersInRange
+                                                .Any(identifier => SymbolEqualityComparer.Default.Equals(identifier.Symbol, semanticModel.GetDeclaredSymbol(variable)))));
+
+        var propertiesInRange = propertiesInClass.Where(property => identifiersInRange
+                                        .Any(identifier => SymbolEqualityComparer.Default.Equals(identifier.Symbol, semanticModel.GetDeclaredSymbol(property))));
+
+        // Now, we iterate through the methods, fields, and properties in the range and extract the necessary information.
+        var pooledMethods = PooledHashSet<MethodSymbolicInfo>.GetInstance();
+        var pooledAttributes = PooledHashSet<AttributeSymbolicInfo>.GetInstance();
+        foreach (var method in methodsInRange)
+        {
+            var parameterTypes = method.ParameterList.Parameters.Count > 0
+                ? method.ParameterList.Parameters
+                    .Where(p => p.Type is not null)
+                    .Select(p => GetFullTypeName(p.Type!, semanticModel))
+                    .ToArray()
+                : Array.Empty<string>();
+
+            pooledMethods.Add(new MethodSymbolicInfo
+            {
+                Name = method.Identifier.Text,
+                ReturnType = GetFullTypeName(method.ReturnType!, semanticModel),
+                ParameterTypes = parameterTypes
+            });
+        }
+
+        var expressionIdentifiersInRange = identifiersInRange.Where(i => i.Identifier.Ancestors().OfType<ExpressionStatementSyntax>().Any());
+        foreach (var field in fieldsInRange)
+        {
+            foreach (var declaredVariable in field.Declaration.Variables)
+            {
+                ExtractAttributeInfo(declaredVariable, field.Declaration.Type, semanticModel, pooledAttributes, writtenInsideSymbols, expressionIdentifiersInRange, cancellationToken);
+            }
+        }
+
+        foreach (var property in propertiesInRange)
+        {
+            ExtractAttributeInfo(property, property.Type, semanticModel, pooledAttributes, writtenInsideSymbols, expressionIdentifiersInRange, cancellationToken);
+        }
+
+        var result = new MemberSymbolicInfo
+        {
+            Methods = pooledMethods.ToArray(),
+            Attributes = pooledAttributes.ToArray()
+        };
+
+        pooledMethods.Free();
+        pooledAttributes.Free();
+
+        return result;
+    }
+
+    private static bool InheritsFromComponentBase(ITypeSymbol componentBaseSymbol, ClassDeclarationSyntax classDeclaration, SemanticModel semanticModel)
+    {
+        if (componentBaseSymbol is null)
+        {
+            return false;
+        }
+
+        var baseTypes = classDeclaration.BaseList?.Types;
+        if (baseTypes is null)
+        {
+            return false;
+        }
+
+        foreach (var baseTypeSyntax in baseTypes)
```

Comment:
This loop seems unnecessary. I think it should be possible to get the symbol info for the class declaration itself (GetDeclaredSymbol I think) and then `InheritsFrom` could be called directly with that type symbol.